### ISSUE TITLE
Gameclient Launch - Gameclient crash on startup process

### ISSUE THREAD

**Issue by Vesorias** [TheCheatsrichter/Gw2_Launchbuddy#261]
Every time I try to launch multiple accounts I get the error: "Client <x> got closed or crashed before a clean Start. Do you want to try to restart this client?". If I say yes, it just gives me the same error. All of my accounts launch just fine individually, but I can't launch more than one at a time which kinda defeats the purpose.

**Comment by Lego6245** [comment]
I've having this problem as well. It appears to put an already existing game window into focus when it happens.

**Comment by NetF1re** [comment]
Do you have injected Software like d3d9.dll in this profile?
I had the same problem and was able to fix it by removing the injected software from the profile or creating a new one without it. Apparently there are problems with shared access.

**Comment by Almdoodler** [comment]
> Do you have injected Software like d3d9.dll in this profile? I had the same problem and was able to fix it by removing the injected software from the profile or creating a new one without it. Apparently there are problems with shared access.

This seems to be a temporary fix, but it would still be great to get this fixed since it worked before and was super nice to not have arc run on every account but only on those where I actually raid/do fractals on.

**Comment by NetF1re** [comment]
With the latest ArcDPS version it works for me again.

**Comment by Almdoodler** [comment]
> With the latest ArcDPS version it works for me again.

Finally had time to test the newest version of arcdps, but it doesn't seem to change anything.  With the newest version injected (of both arcdps ( and arcdps boon table) I'm unable to launch the game at all when using LB. 

It only starts when not injecting arcdps and the boon table addon or after restarting LB and repairing the Loginfile for what seems like a random number of times. If the second is the case it only works for one time launching of the game, if I close it and want to start it again, I have to do the whole process all over again.

**Comment by Master-Liu-TBF** [comment]
I've been havingthe same issue happening, I was using the 2.4.1 and LB wasnt updating even though there was a newer version of 2.4.2 it hadn't been checking even when I would click on 'Check for updates'. When I went to the latest version of 2.4.2 and deleted all my accounts and re-entered them it worked fine, I was able to open multiple gw2 accounts. But after the initial opening it's back to doing same thing as before and erroring, "Client got closed or crashed before a clean Start. Do you want to try to restart this client?".

And when I added Taco and BlishHud, I can only get BlishHud to work. Taco doesnt seem to launch via LB. I have no idea what happened, I haven't changed anything on my computer. I've had to download another gw2 launcher to workaround this error. Though I quite enjoy LB, I can only launch one account, I can not have more then one open. 

Hopefully this will get resolved.

**Comment by YellowGB** [comment]
I have exactly the same issue.
No 3rd party or injected software whatsoever, I updated to 2.4.2 but still have the issue. I don't understand because I didn't change anything in any settings and it was launching 5 accounts simultaneously just fine one week ago, but now I have to launch them one by one or I get this error, which kinda defeats the purpose of LB ^^
If anyone has the solution to this, I'll gladly take it :)

**Comment by YellowGB** [comment]
Ok, after rebooting my computer a second time, I no longer have this issue. I installed nodejs on my computer the other day for a project I'm working on, I wonder if its not after that that I started to have problems with LB. I don't see how nodejs would impact this but that's the only link I can think of.

**Comment by Master-Liu-TBF** [comment]
> Ok, after rebooting my computer a second time, I no longer have this issue. I installed nodejs on my computer the other day for a project I'm working on, I wonder if its not after that that I started to have problems with LB. I don't see how nodejs would impact this but that's the only link I can think of.

Unfortunately, I don't have other projects or anything like that. Nothing changed on my system, not even a reboot remendied the problem. But I'm glad that you found a solution.

**Comment by YellowGB** [comment]
> > Ok, after rebooting my computer a second time, I no longer have this issue. I installed nodejs on my computer the other day for a project I'm working on, I wonder if its not after that that I started to have problems with LB. I don't see how nodejs would impact this but that's the only link I can think of.
> 
> Unfortunately, I don't have other projects or anything like that. Nothing changed on my system, not even a reboot remendied the problem. But I'm glad that you found a solution.

Maybe I just got lucky, I'm afraid to turn off my computer now ^^
I see @TheCheatsrichter has put this issue in testing, maybe he will find what's causing this.

**Comment by Master-Liu-TBF** [comment]
> > > Ok, after rebooting my computer a second time, I no longer have this issue. I installed nodejs on my computer the other day for a project I'm working on, I wonder if its not after that that I started to have problems with LB. I don't see how nodejs would impact this but that's the only link I can think of.
> > 
> > 
> > Unfortunately, I don't have other projects or anything like that. Nothing changed on my system, not even a reboot remendied the problem. But I'm glad that you found a solution.
> 
> Maybe I just got lucky, I'm afraid to turn off my computer now ^^ I see @TheCheatsrichter has put this issue in testing, maybe he will find what's causing this.

I've had the normal windows updates, but that's about it. But yeah, I get it, I would afraid aswell to turn off my comp :)

**Comment by TheCheatsrichter** [comment]
Here a quick overview of the client launch process:
https://github.com/TheCheatsrichter/Gw2LBPluginTemplates/wiki/Launchbuddy-Client-Status-Diagram

I think the main problem here is that the errormessage is to generalized.
Pushed a better Exception handler with commit https://github.com/TheCheatsrichter/Gw2_Launchbuddy/commit/9b22b7a4c7039d73d4f6f463f5b953d393f28bf9
this should now provide a unique errorcode to pinpoint the cleintlaunch step in which the gameclient crashed.

Lego6245:
His issue seems to be a mutex problem. If an allready existing gw2 instance is focused it is very likely that the mutex wasn't closed correctly --> LB expects a new process ID, however the new instance of the gameclient was never launched --> looks like a  gameclient crash to LB.

Almdoodler + Netfire:
3rd party .dll injection seem to have caused the crash. Incompatible .dlls crash the gameclient in an early stage --> looks like a  gameclient crash to LB.

YellowGB + Master-Liu_TBF:
I am currently not able to replicate the behavior of your two usecases. I will keep an eye on the launchprocess when Version 3.0 alphatesting starts.

**Comment by YellowGB** [comment]
@TheCheatsrichter Thank you for your answers, just for you information it's happening again tonight : 
![image](https://user-images.githubusercontent.com/82955894/151626378-929d2923-8d92-481d-8b7f-b98cc760418d.png)

I was able to launch 3 sessions simultaneously, but when I tried to launch a 4th one (I usually launch 5 together at a time because I have many accounts), this error message appeared again. And now I can't launch any game simulatenously again.

I'm looking forward for the new 3.0 version, and thank you for your work by the way, it's a really useful piece of software you made here, just started to have problems with it recently but I've been enjoying it for years :)

**Comment by Master-Liu-TBF** [comment]
> @TheCheatsrichter Thank you for your answers, just for you information it's happening again tonight : ![image](https://user-images.githubusercontent.com/82955894/151626378-929d2923-8d92-481d-8b7f-b98cc760418d.png)
> 
> I was able to launch 3 sessions simultaneously, but when I tried to launch a 4th one (I usually launch 5 together at a time because I have many accounts), this error message appeared again. And now I can't launch any game simulatenously again.
> 
> I'm looking forward for the new 3.0 version, and thank you for your work by the way, it's a really useful piece of software you made here, just started to have problems with it recently but I've been enjoying it for years :)

For me it was if I wanted to launch more then 1, I would launch my main acct, then try a 2nd or 3rd, I never launched more then one at a time and I would still get the error after the main/1st. I deleted the accounts and then added them back in added my gfx settings which hasn't changed. It'll work once, and that's it. Can't launch beyond the first one, if I exit the first then I can launch either the 2nd or the 3rd but can not launch while an account is launched already, I get the error as stated.

Look forward to 3.0.

Thanks

**Comment by marcusklausen** [comment]
Hi, just wanted to chime in, had this issue and seem to have fixed it.
I was by mistake injecting `dxgi.dll` and `d3d11.dll`. I'm pretty sure `Gw2-64.exe` needs `d3d11.dll`, if you have dx11 enabled, which as of now it is by default. Removed all dlls and added back only `addonLoader.dll`, which is the one provided by AddonManager, and it hooks every addon by itself, so no need for any other dlls to be added. 

As a side note I can only speak for arcdps and its siblings and then gw2radial, cant speak for reshade etc.

Will correct myself here, if it starts failing again, but I've booted the game a couple of times and it has worked every time compared to 1/20 times before.
